flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -237,8 +254,15 @@ public boolean isEmpty() {
 						computationState.getCounter());
 
 				newComputationStates = Collections.emptyList();
+				nfaChanged = true;
```

Comment:
I think this may not always be true. For example, the `eventSharedBuffer.release`(NFA:250) may already have released the events that will be pruned. This will cause the pruning sharedBuffer do nothing. Thoughts?
Anyway, leave it here will be more safe.:)